appinventor/components/src/com/google/appinventor/components/runtime/RecyclerView.java

```diff
@@ -0,0 +1,721 @@
+      txtSearchBox.setVisibility(View.VISIBLE);
+    } else {
+      txtSearchBox.setVisibility(View.GONE);
+    }
+
+    // set the colors and initialize the elements
+    // note that the TextColor and ElementsFromString setters
+    // need to have the textColor set first, since they reset the
+    // adapter
+
+    Width(Component.LENGTH_FILL_PARENT);
+    BackgroundColor(DEFAULT_BACKGROUND_COLOR);
+    SelectionColor(DEFAULT_SELECTION_COLOR);
+    ImageWidth(DEFAULT_IMAGE_WIDTH);
+    ImageHeight(DEFAULT_IMAGE_WIDTH);
+  
+    textColor = DEFAULT_TEXT_COLOR;
+    detailTextColor = DEFAULT_TEXT_COLOR;
+    textSize = DEFAULT_TEXT_SIZE;
+    detailTextSize = DEFAULT_TEXT_SIZE;
+    gridCount=DEFAULT_GRID_COUNT;
+
+
+    
+    linearLayout.addView(txtSearchBox);
+    linearLayout.addView(recyclerView);
+    linearLayout.requestLayout();
+    //recyclerView.requestLayout();
+    container.$add(this);
+};
+
+  @Override
+  public View getView() {
+    return linearLayout;
+  }
+
+  /**
+  * Sets the height of the listView on the screen
+  * @param height for height length
+  */  
+  @Override
+  @SimpleProperty(description = "Determines the height of the list on the view.",
+      category =PropertyCategory.APPEARANCE)
+  public void Height(int height) {
+    if (height == LENGTH_PREFERRED) {
+      height = LENGTH_FILL_PARENT;
+    }
+    super.Height(height);
+  }
+
+  /**
+  * Sets the width of the listView on the screen
+  * @param width for width length
+  */
+  @Override
+  @SimpleProperty(description = "Determines the width of the list on the view.",
+      category = PropertyCategory.APPEARANCE)
+  public void Width(int width) {
+    if (width == LENGTH_PREFERRED) {
+      width = LENGTH_FILL_PARENT;
+    }
+    super.Width(width);
+  }
+
+ /**
+   * Sets true or false to determine whether the search filter box is displayed in the ListView
+   *
+   * @param showFilter set the visibility according to this input
+  */ 
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+      defaultValue = DEFAULT_ENABLED ? "True" : "False")
+  @SimpleProperty(description = "Sets visibility of ShowFilterBar. True will show the bar, " +
+      "False will hide it.")
+  public void ShowFilterBar(boolean showFilter) {
+    this.showFilter = showFilter;
+    if (showFilter) {
+      txtSearchBox.setVisibility(View.VISIBLE);
+    }
+    else {
+      txtSearchBox.setVisibility(View.GONE);
+    }
+  }
+
+  /**
+   * Returns true or false depending on the visibility of the Filter bar element
+   * @return true or false (visibility)
+  */ 
+  @SimpleProperty(category = PropertyCategory.BEHAVIOR,
+      description = "Returns current state of ShowFilterBar for visibility.")
+  public boolean ShowFilterBar() {
+    return showFilter;
+  }
+
+ public void setAdapterr(){
+
+    int size = currentItems.size();
+
+    String[] first=new String[size];
+    String[] second=new String[size];
+    boolean[] select=new boolean[size];
```

Comment:
Select is probably a bad name for a variable. It's not a keyword in java, but it is in SQL and other things java might interact with. So it doesn't seem to be causing trouble, but I'd rename it anyway.

Selected or selections seems fine.